### DISCUSSION TITLE
GitHub Classroom + GitHub Desktop is causing issues (related to private remote 'upstream')

### DISCUSSION THREAD

**Discussion by lenntt** [community/community#145651] [GitHub Education]
### Hi everyone,

# Problem Context
- I'm using github classrooms for assignments and since our students are not familiarize with git the first semester, we get them started with github desktop.
- GitHub classrooms uses my own template repository (which is private!) as the upstream and every students its own FORK - (i think recently changed implementation?) .  
- Github desktop, when cloning a fork, adds the fork locally as remote 'origin', but also the remote 'upstream'.
    - and: it keeps adding the upstream every time github desktop opens, even after a manual `git remote rm upstream`.
When syncing, it also tries fetching from upstream.

# Problem
The problem, is that the template repository is private, so whenever they fetch, it results in a nasty error that's hard to explain to students new to git.
After the classroom is created, I can't switch the visibility to public (it gives a different error) - which is also not desired - I'd like to have only the people in, who're in the class.
The alternative, is having to manage the people in there manually - doesn't seem very productive either.

# Desired behavior
Desired behavior, either:
- GitHub desktop not to automatically add `upstream` - or at least the ability to switch it off
- automatically give `upstream` read access to students who accepted the classroom (preferred).

# Remarks
- I think the current implementation of classroom, breaks one of the concept of github: forking a repository you cant see seems not a supported use case.

PS: I hope I dont come across being rude, but I'm getting frustrated.  this is the 3rd time I'm trying explain a GitHub issue ON github and TO GitHub. Other reports are: 
- https://github.com/community/Global-Campus-Teachers/discussions/380 
- Ticket 3103139 (not public) -> outcome: without acknowledgement, asked me to report in this community (... so please dont point me somewhere else)

**Comment by dcowell** [comment]
We are experiencing this issue, again not for all users, and its really creating problems. The problem looks to be described here as well.

https://github.com/desktop/desktop/issues/3715
